pmg/bills.py

```diff
@@ -111,9 +111,61 @@ def bill_history(bill):
         info.update(location)
         history.append(info)
 
+    history = hansard_linking(history)
     return history
 
 
+def match_title(event_title):
+    """
+    Match bill title against the following possible titles
+    """
+    bill_titles = [
+        "Bill passed by National Assembly",
+        "Bill passed by both Houses",
+        "Bill revived on this date",
+        "The NCOP rescinded",
+        "Bill remitted",
+    ]
+    for title in bill_titles:
+        if title in event_title:
+            return True
+    return False
+
+
+def match_dates(hansard_date, event_date):
+    hansard_iso_date = iso8601.parse_date(hansard_date)
+    event_iso_date = iso8601.parse_date(event_date)
+    if hansard_iso_date.date() == event_iso_date.date():
+        return True
+    return False
+
+
+def hansard_linking(bill_history):
+    """
+    We need to link certain bill events to hansards
+    Hansrds will always be linked to a house (NA or NCOP)
+    The Date of the bill event and the hansard will be the same.
+    Bill Titles we are looking for:
+    * Bill passed by National Assembly
+    * Bill passed by both Houses
+    * Bill revived on this date
+    * The NCOP rescinded
+    * Bill remitted
+    """
+
+    for class_history in bill_history:
```

Comment:
I would say it is important to unit test this - 7 levels of nesting is not something that's quick to verify by eye, but writing the tests should be really easy.

Reply:
I think an integration test where this doesn't have any effect and another where it does would also be in order, so that the thread where this works on objects coming from the API for reals isn't accidentally broken. Doesn't need to check detail logic, just enough to be sure that the integration isn't broken.